File: app/src/main/java/beans/prj/newsapp/di/ActivityBuilder.java
package beans.prj.newsapp.di;

import beans.prj.newsapp.ui.splash.SplashActivity;
import dagger.Module;
import dagger.android.ContributesAndroidInjector;

@Module
public abstract class ActivityBuilder {

    @ContributesAndroidInjector
    abstract SplashActivity bindSplashActivity();

}

File: app/src/main/java/beans/prj/newsapp/di/ActivityBuilder.kt
package beans.prj.newsapp.di

import beans.prj.newsapp.ui.home.HomeActivity
import beans.prj.newsapp.ui.splash.SplashActivity
import dagger.Module
import dagger.android.ContributesAndroidInjector

@Module
abstract class ActivityBuilder {

    @ContributesAndroidInjector
    abstract fun bindSplashActivity(): SplashActivity

    @ContributesAndroidInjector
    abstract fun bindHomeActivity(): HomeActivity
}